mlflow/models/resources.py

```diff
@@ -0,0 +1,145 @@
+        """
+        Convert the resource to a dictionary.
+        Subclasses must implement this method.
+        """
+
+
+@dataclass
+class DatabricksResource(Resource, ABC):
+    """
+    Base class to define all the Databricks resources to serve a model.
+    """
+
+    target_uri: Providers = Providers.DATABRICKS.value
+
+
+@dataclass
+class DatabricksLLMEndpoint(DatabricksResource):
+    """
+    Define Databricks LLM endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks endpoints used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_LLM_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksChatEndpoint(DatabricksResource):
+    """
+    Define Databricks LLM endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks chat endpoints
+        used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_CHAT_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksEmbeddingEndpoint(DatabricksResource):
+    """
+    Define Databricks embedding endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks embedding endpoints
+        used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_EMBEDDINGS_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksVectorSearchEndpoint(DatabricksResource):
+    """
+    Define Databricks vector search endpoint resource to serve a model.
+
+    Args:
+        endpoint_name (str): The name of all the databricks vector search endpoints
+        used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_VECTOR_SEARCH_ENDPOINT_NAME
+    endpoint_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.endpoint_name]} if self.endpoint_name else {}
+
+
+@dataclass
+class DatabricksVectorSearchIndexName(DatabricksResource):
+    """
+    Define Databricks vector search index name resource to serve a model.
+
+    Args:
+        index_name (str): The name of all the databricks vector search index names
+        used by the model.
+    """
+
+    type: ResourceType = ResourceType.DATABRICKS_VECTOR_SEARCH_INDEX_NAME
+    index_name: str = None
+
+    def to_dict(self):
+        return {self.type.value: [self.index_name]} if self.index_name else {}
+
+
+class _ResourceBuilder:
+    """
+    Private builder class to build the resources dictionary.
+    """
+
+    @staticmethod
+    def from_resources(resources: List[Resource]) -> dict:
```

Comment:
We need to offer a public API for this, right?

Reply:
Not really.  

Users would ideally do this:
```
mlflow.pyfunc.log_model(...,
    resources=[DatabricksVectorSearchEndpoint(...), DatabricksVectorSearchIndex(...),DatabricksVectorSearchIndex(...)]
)
or 

mlflow.pyfunc.log_model(...,
    resources="file_path_resource"
)
```

So we will use this API internally only. I have updated the design doc with that. Let me know if this approach makes sense.